### DISCUSSION TITLE
limitations around `fish_user_key_bindings` function

### DISCUSSION THREAD

**Discussion by ZuBB** [fish-shell/fish-shell#9953] [Q&A]
Are there any limitations on content within `fish_user_key_bindings` function? Amount of bindings? Order? Anything else?

I have two different aliases, and one works differently when they are together in a single file. When in separate files -- all is OK.

```sh
function fish_user_key_bindings
  bind \cw backward-kill-bigword
  # https://github.com/fish-shell/fish-shell/issues/2904                                                                                  
  #bind \cc 'echo; commandline | cat; commandline ""; commandline -f repaint'
end
```

Any hint of what is going wrong?

**Comment by faho** [comment]
> Are there any limitations on content within fish_user_key_bindings function? Amount of bindings?

There is no specific limitation, no.

It's just a function and can be as long as your computer's memory allows.

In practical terms it should be unlimited.

> Order? 

The last binding for a given sequence wins.

> I have two different aliases

You don't show any aliases here.

This contains one binding and one commented binding (because of the leading "#").

> one works differently when they are together in a single file

How? What does it do differently?

Is there any other definition of it anywhere?

Autoloading only happens if the function isn't already defined, and otherwise the last executed definition of any function wins.

> Any hint of what is going wrong?

I believe you are misidentifying the problem and we need more context and explanation.

**Reply by faho** [reply]
Alright I have checked [your dotfiles](https://github.com/ZuBB/dotfiles/tree/master/.config/fish) and I can't see what you are talking about. I don't understand your description of your issue and would need more explanation.

I *can* tell you that the entirety of https://github.com/ZuBB/dotfiles/blob/master/.config/fish/conf.d/zzz-021_when_interactive.fish is unnecessary because fish already disables ixon, by default.

**Reply by ZuBB** [reply]
I am stuck a bit with personal things, I will try to provide a better description and some visuals that demonstrate the issue.

Thanks a lot for taking this proactively
